### PR TITLE
Nucleotide, Rnase, Digestion Params Clone Method, and Interface Modificaiton

### PR DESCRIPTION
**Pull request by nbollis**
Adjusted interfaces to allow for abstraction of the SpectralMatchClass in MetaMorpheus

Added Clone method to DigestionParams
  There are many many times in MetaMorpheus where we need to copy digestion parameters to file specific parameters and they are done manually each time. Once incorporated, will allow all cloning to take place in one location

Added Nucleotide class that is modeled off of the Residue Class

Added Rnase without digestion functionality. 
This will make subsequent PR's with digestion easier

### CODE REVIEW

File: mzLib/Omics/IBioPolymer.cs [hunk 1]
```diff
@@ -1,24 +1,25 @@
...
+        string FullName { get; }
         string BaseSequence { get; }
         int Length { get; }
         string DatabaseFilePath { get; }
         bool IsDecoy { get; }
         bool IsContaminant { get; }
         string Organism { get; }
         string Accession { get; }
+        /// <summary>
+        /// The list of gene names consists of tuples, where Item1 is the type of gene name, and Item2 is the nameof the specific molecule.
```
Comment (trishorts) [inline mzLib/Omics/IBioPolymer.cs:18]: https://www.uniprot.org/help/gene_name

Comment (trishorts) [reply mzLib/Omics/IBioPolymer.cs:18]: image

Comment (trishorts) [reply mzLib/Omics/IBioPolymer.cs:18]: when reading uniprot xml, the two fields that go in the tuple in this case are "primary" and "dpiA", which is the most common gene symbol. I believe that the other pairs of words are added as the tuple in the list. I'm not sure what is done with those values.

Comment (trishorts) [reply mzLib/Omics/IBioPolymer.cs:18]: Here is my attempt at a comment:
//This tuple originates from reading UniProt XML protein entries. It consists of two parts.
//The second part is the gene name, which is also often referred to as the gene symbol.
//The first part is a descriptor for the second part. For example, "primary" is a word used for the 
//first part to indicate that the gene symbol/name that follows is the name most widely accepted
//by the community. The word "synonym" in the first part would indicate that the gene name in the
//second part is an alternative. The word "ORF", which stands for open reading frame in the first part
//is also acceptable. Here, the second part will include a range, for example "AO055_05240"

**Review by trishorts — APPROVED** [review]
check the comments

**Review by trishorts — APPROVED** [review]

**Review by elaboy — APPROVED** [review]

**Review by Alexander-Sol — APPROVED** [review]

**Pull request merged**